### PR TITLE
Improve the  error message when resetting the password

### PR DESCRIPTION
**Pull request by Kanapriya**
## Purpose
Resolves : https://github.com/wso2/product-is/issues/7761

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java [hunk 1]
```diff
@@ -346,8 +347,9 @@ private boolean isPasswordPolicyViolationError(Throwable e) {
         while (e != null) {
             if (e.getCause() instanceof PolicyViolationException) {
                 return true;
-            }
-            if (e.getCause() instanceof IdentityPasswordHistoryException) {
+            } else if (e.getCause() instanceof IdentityPasswordHistoryException) {
+                return true;
+            } else if (e.getCause() instanceof IdentityEventException) {
```
Comment (IsuraD) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java]: Why we treat IdentityEventException as PasswordPolicyViolationError?

Comment (Kanapriya) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java]: Fixed this with specified error code in https://github.com/wso2-extensions/identity-outbound-auth-passwordPolicy/pull/30/commits/610c8fe39a202097028cc7479ce9e28ee57113aa

File: component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java [hunk 2]
```diff
@@ -346,8 +347,12 @@ private boolean isPasswordPolicyViolationError(Throwable e) {
         while (e != null) {
             if (e.getCause() instanceof PolicyViolationException) {
                 return true;
-            }
-            if (e.getCause() instanceof IdentityPasswordHistoryException) {
+            } else if (e.getCause() instanceof IdentityPasswordHistoryException) {
+                return true;
+            } else if (e.getCause() instanceof IdentityEventException &&
```
Comment (IsuraD) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java]: can simplyfy as bellow

 else if (e.getCause() instanceof IdentityEventException && PasswordPolicyConstants.PASSWORD_HISTORY_VIOLATION_ERROR_CODE.equals(I(dentityEventException) e.getCause()).getErrorCode())) {

Comment (Kanapriya) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java]: fix in https://github.com/wso2-extensions/identity-outbound-auth-passwordPolicy/pull/30/commits/6051a95e02b102f6e5f40a8958a35e7f02b61805

**Pull request merged**